submissions/Ant-C-tech/oop-exercise/index.js

```diff
@@ -0,0 +1,113 @@
+/* Refer to https://github.com/OleksiyRudenko/a-tiny-JS-world for the task details
+   Complete the below for code reviewers' convenience:
+
+   Code repository: https://github.com/Ant-C-tech/a-tiny-JS-world
+   Web app: _put project's github pages URL here_
+   */
+
+// ======== OBJECTS DEFINITIONS ========
+// Define your objects here
+
+class Inhabitant {
+    constructor(species, name, legs, saying, friends) {
+        this.species = species;
+        this.name = name;
+        this.legs = legs;
+        this.friends = friends;
+        this.saying = saying;
+
+        this.templete = [`I am a ${this.species}`,
+            `My name is ${ this.name }`,
+            `I have ${this.legs} legs`,
+            `I want to say you: "${this.saying}"`
+        ].join('. ');
```

Comment:
Not illegal but there may be an issue.
Imagine a property (e.g. `saying`) gets changed during the object life cycle and we want this change reflected when `getVoice` gets called next time.

`friends` prop has this privilege. Any good reason to handle friends and the group of "simpler" props differently?

Answer to this question leads to a proper approach.